cpp/include/Ice/Proxy.h

```diff
@@ -1186,19 +815,19 @@ class ICE_API ObjectPrx : public Proxy<ObjectPrx>, public ::std::enable_shared_f
     // This constructor is never called; it allows Proxy's default constructor to compile.
     ObjectPrx() = default;
 
-    template<typename R, template<typename> class P = ::std::promise, typename Obj, typename Fn, typename... Args>
+    template<typename R, template<typename> class P = std::promise, typename Obj, typename Fn, typename... Args>
```

Comment:
I think it would be cleaner to move _makePromiseOutgoing and _makeLambdaOutgoing out of ObjectPrx ... for another PR.